The job **`test-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/10239317952/job/28324775839) [(plain)](https://github.com/rust-lang-ci/rust/commit/df62a424cdfadcdfbeb852c63096dccfc711c032/checks/28324775839/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [codegen] tests/codegen/issues/issue-118306.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/issues/issue-118306/issue-118306.ll" "/checkout/tests/codegen/issues/issue-118306.rs" "--check-prefix=CHECK" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
/checkout/tests/codegen/issues/issue-118306.rs:13:17: error: CHECK-NEXT: expected string not found in input
/checkout/tests/codegen/issues/issue-118306.rs:13:17: error: CHECK-NEXT: expected string not found in input
 // CHECK-NEXT: [[_2:%.*]] = and i64 [[INPUT:%.*]], 3
                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/issues/issue-118306/issue-118306.ll:10:7: note: scanning from here
      ^
      ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/issues/issue-118306/issue-118306.ll:12:2: note: possible intended match here
 %1 = and i32 %0, 3

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/issues/issue-118306/issue-118306.ll
Check file: /checkout/tests/codegen/issues/issue-118306.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
           1: ; ModuleID = 'issue_118306.9d0af450ab9e47e7-cgu.0' 
           2: source_filename = "issue_118306.9d0af450ab9e47e7-cgu.0" 
           3: target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-n32:64-S128-ni:1:10:20" 
           4: target triple = "wasm32-unknown-wasi" 
           5:  
           6: @switch.table.branchy = private unnamed_addr constant [4 x i64] [i64 0, i64 1, i64 1, i64 2], align 8 
           7:  
           8: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) 
           9: define dso_local noundef range(i64 0, 3) i64 @branchy(i64 noundef %input) unnamed_addr #0 { 
          10: start: 
next:13'0           X error: no match found
          11:  %0 = trunc i64 %input to i32 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          12:  %1 = and i32 %0, 3 
next:13'0     ~~~~~~~~~~~~~~~~~~~~
next:13'1      ?                   possible intended match
          13:  %switch.gep = getelementptr inbounds [4 x i64], ptr @switch.table.branchy, i32 0, i32 %1 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          14:  %switch.load = load i64, ptr %switch.gep, align 8 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          15:  ret i64 %switch.load 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~
          16: } 
next:13'0     ~~
          17:  
next:13'0     ~
          18: attributes #0 = { mustprogress nofree norecurse nosync nounwind willreturn memory(none) "target-cpu"="generic" } 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          19:  
next:13'0     ~
          20: !llvm.ident = !{!0} 
next:13'0     ~~~~~~~~~~~~~~~~~~~~
          21:  
next:13'0     ~
          22: !0 = !{!"rustc version 1.82.0-nightly (df62a424c 2024-08-04)"} 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

--- a/tests/codegen/issues/issue-107681-unwrap_unchecked.rs
+++ b/tests/codegen/issues/issue-107681-unwrap_unchecked.rs
@@ -1,0 +1,20 @@
+//@ compile-flags: -O
+//@ min-llvm-version: 19
+
+// Test for #107681.
+// Make sure we don't create `br` or `select` instructions.
+
+#![crate_type = "lib"]
+
+use std::iter::Copied;
+use std::slice::Iter;
+
+#[no_mangle]
+pub unsafe fn foo(x: &mut Copied<Iter<'_, u32>>) -> u32 {
+    // CHECK-LABEL: @foo(
+    // CHECK-NOT: br
+    // CHECK-NOT: select
+    // CHECK: [[RET:%.*]] = load i32, ptr
+    // CHECK-NEXT: ret i32 [[RET]]
+    x.next().unwrap_unchecked()
+}

--- a/tests/codegen/issues/issue-118306.rs
+++ b/tests/codegen/issues/issue-118306.rs
@@ -1,0 +1,23 @@
+//@ compile-flags: -O
+//@ min-llvm-version: 19
+//@ only-x86_64
+
+// Test for #118306.
+// Make sure we don't create `br` or `select` instructions.
+
+#![crate_type = "lib"]
+
+#[no_mangle]
+pub fn branchy(input: u64) -> u64 {
+    // CHECK-LABEL: @branchy(
+    // CHECK-NEXT:  start:
+    // CHECK-NEXT:    [[_2:%.*]] = and i64 [[INPUT:%.*]], 3
+    // CHECK-NEXT:    [[SWITCH_GEP:%.*]] = getelementptr inbounds [4 x i64], ptr @switch.table.branchy, i64 0, i64 [[_2]]
+    // CHECK-NEXT:    [[SWITCH_LOAD:%.*]] = load i64, ptr [[SWITCH_GEP]]
+    // CHECK-NEXT:    ret i64 [[SWITCH_LOAD]]
+    match input % 4 {
+        1 | 2 => 1,
+        3 => 2,
+        _ => 0,
+    }
+}

--- a/tests/codegen/issues/issue-126585.rs
+++ b/tests/codegen/issues/issue-126585.rs
@@ -1,0 +1,24 @@
+//@ compile-flags: -Copt-level=s
+//@ min-llvm-version: 19
+//@ only-x86_64
+
+// Test for #126585.
+// Ensure that this IR doesn't have extra undef phi input, which also guarantees that this asm
+// doesn't have subsequent labels and unnecessary `jmp` instructions.
+
+#![crate_type = "lib"]
+
+#[no_mangle]
+fn checked_div_round(a: u64, b: u64) -> Option<u64> {
+    // CHECK-LABEL: @checked_div_round
+    // CHECK: phi
+    // CHECK-NOT: undef
+    // CHECK: phi
+    // CHECK-NOT: undef
+    match b {
+        0 => None,
+        1 => Some(a),
+        // `a / b` is computable and `(a % b) * 2` can not overflow since `b >= 2`.
+        b => Some(a / b + if (a % b) * 2 >= b { 1 } else { 0 }),
+    }
+}

